src/resdk/resources/geneset.py

```diff
@@ -0,0 +1,118 @@
+"""Geneset resource."""
+import logging
+from collections.abc import Set
+
+from .data import Data
+
+
+class BaseGeneset(Set):
+    """Base class for Geneset resource.
+
+    :param genes: genes
+    :type genes: list of strings
+    """
+
+    def __init__(self, genes):
+        """Initialize genes list."""
+        self.genes = []
+        self.init(genes)
+
+    def init(self, genes):
+        """Initialize genes list.
+
+        :param genes: list
+        """
+        self.genes = []
+        if genes is not None:
+            for value in genes:
+                if value not in self.genes:
+                    self.genes.append(value)
+
+    def __iter__(self):
+        """Iterate genes."""
+        return iter(self.genes)
+
+    def __contains__(self, gene):
+        """Check if current genes list contains gene."""
+        return gene in self.genes
+
+    def __len__(self):
+        """Return number of genes."""
+        return len(self.genes)
+
+
+class Geneset(Data):
+    """Resolwe Geneset resource.
+
+    :param resolwe: Resolwe instance
+    :type resolwe: Resolwe object
+    :param model_data: Resource model data
+
+    """
+
+    WRITABLE_FIELDS = Data.WRITABLE_FIELDS + (
+        "genes",
+        "source",
+        "species",
+    )
+
+    def __init__(self, resolwe, **model_data):
+        """Initialize attributes."""
+        self.logger = logging.getLogger(__name__)
+
+        self._genes = None
+        self.source = None
+        self.species = None
+
+        super().__init__(resolwe, **model_data)
+
+    @property
+    def genes(self):

```

Comment:
This should go on the server and get the genes. Use this code to fetch them:
```
url = urljoin(self.resolwe.url, "api", "storage", self.output["geneset_json"])
response =  self.resolwe.session.get(url)
genes = json.loads(response.content)["json"]["genes"]
```

Reply:
I used that code now for fetching genes. I hope it is done the right way 😄